Fix failing doc tests

Tests will now be run against PRs, but we had to disable `doctest` in order to make them pass. This should ideally be fixed so we can reenable them.

<details>
  <summary>Here are the current failures</summary>

```sh
---- src/lib.rs - (line 115) stdout ----
error: expected expression, found `$`
 --> src/lib.rs:116:1
  |
3 | $ export AWS_ACCESS_KEY_ID=AKIAIOSFODNN7EXAMPLE
  | ^ expected expression

error: aborting due to previous error

Couldn't compile the test.
---- src/lib.rs - (line 103) stdout ----
error: expected `;`, found `aws_access_key_id`
 --> src/lib.rs:104:10
  |
3 | [default]
  |          ^ help: add `;` here
4 | aws_access_key_id=AKIAIOSFODNN7EXAMPLE
  | ----------------- unexpected token

error: expected `;`, found `aws_secret_access_key`
 --> src/lib.rs:105:39
  |
4 | aws_access_key_id=AKIAIOSFODNN7EXAMPLE
  |                                       ^ help: add `;` here
5 | aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  | --------------------- unexpected token

error[E0425]: cannot find value `aws_secret_access_key` in this scope
 --> src/lib.rs:106:1
  |
5 | aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  | ^^^^^^^^^^^^^^^^^^^^^
  |
help: you might have meant to introduce a new binding
  |
5 | let aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  | +++

error[E0425]: cannot find value `wJalrXUtnFEMI` in this scope
 --> src/lib.rs:106:23
  |
5 | aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  |                       ^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `K7MDENG` in this scope
 --> src/lib.rs:106:37
  |
5 | aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  |                                     ^^^^^^^ not found in this scope

error[E0425]: cannot find value `bPxRfiCYEXAMPLEKEY` in this scope
 --> src/lib.rs:106:45
  |
5 | aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
  |                                             ^^^^^^^^^^^^^^^^^^ not found in this scope

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0425`.
Couldn't compile the test.
---- src/types.rs - types::LakeConfigBuilder::mainnet (line 61) stdout ----
error[E0752]: `main` function is not allowed to be `async`
 --> src/types.rs:64:1
  |
5 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0752`.
Couldn't compile the test.
---- src/lib.rs - streamer (line 241) stdout ----
error[E0596]: cannot borrow `stream` as mutable, as it is not declared as mutable
  --> src/lib.rs:254:40
   |
13 |     let (_, stream) = near_lake_framework::streamer(config);
   |             ------ help: consider changing this to be mutable: `mut stream`
14 |
15 |     while let Some(streamer_message) = stream.recv().await {
   |                                        ^^^^^^^^^^^^^ cannot borrow as mutable

error[E0752]: `main` function is not allowed to be `async`
 --> src/lib.rs:245:1
  |
6 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0596, E0752.
For more information about an error, try `rustc --explain E0596`.
Couldn't compile the test.
---- src/types.rs - types::LakeConfigBuilder::betanet (line 97) stdout ----
error[E0752]: `main` function is not allowed to be `async`
 --> src/types.rs:100:1
  |
5 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0752`.
Couldn't compile the test.
---- src/types.rs - types::LakeConfig (line 7) stdout ----
error[E0752]: `main` function is not allowed to be `async`
 --> src/types.rs:10:1
  |
5 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0752`.
Couldn't compile the test.
---- src/lib.rs - (line 73) stdout ----
error[E0752]: `main` function is not allowed to be `async`
 --> src/lib.rs:76:1
  |
5 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0752`.
Couldn't compile the test.
---- src/lib.rs - (line 7) stdout ----
error[E0308]: mismatched types
   --> src/lib.rs:33:4
    |
7   |   async fn main() -> Result<(), tokio::io::Error> {
    |                      ---------------------------- expected `Result<(), std::io::Error>` because of return type
...
28  | /    match sender.await {
29  | |        Ok(Ok(())) => Ok(()),
30  | |        Ok(Err(e)) => Err(e),
31  | |        Err(e) => Err(anyhow::Error::from(e)), // JoinError
32  | |    }
    | |____^ expected struct `std::io::Error`, found struct `anyhow::Error`
    |
    = note: struct `anyhow::Error` and struct `std::io::Error` have similar names, but are actually distinct types
note: struct `anyhow::Error` is defined in crate `anyhow`
   --> /Users/morganmccauley/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.65/src/lib.rs:371:1
    |
371 | pub struct Error {
    | ^^^^^^^^^^^^^^^^
note: struct `std::io::Error` is defined in crate `std`
   --> /Users/morganmccauley/.rustup/toolchains/1.67-aarch64-apple-darwin/lib/rustlib/src/rust/library/std/src/io/error.rs:68:1
    |
68  | pub struct Error {
    | ^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
Couldn't compile the test.
---- src/types.rs - types::LakeConfig::s3_config (line 32) stdout ----
error[E0432]: unresolved import `http`
 --> src/types.rs:34:5
  |
4 | use http::Uri;
  |     ^^^^ use of undeclared crate or module `http`

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
 --> src/types.rs:33:17
  |
3 | use aws_sdk_s3::Endpoint;
  |                 ^^^^^^^^
  |
  = note: `#[warn(deprecated)]` on by default

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
  --> src/types.rs:42:13
   |
12 |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
   |             ^^^^^^^^

error[E0277]: the trait bound `Result<Endpoint, InvalidEndpointError>: ResolveEndpoint<Params>` is not satisfied
   --> src/types.rs:42:13
    |
11  |         .endpoint_resolver(
    |          ----------------- required by a bound introduced by this call
12  |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ResolveEndpoint<Params>` is not imple
mented for `Result<Endpoint, InvalidEndpointError>`
    |
    = help: the following other types implement trait `ResolveEndpoint<Params>`:
              <&'static str as ResolveEndpoint<T>>
              <DefaultResolver as ResolveEndpoint<Params>>
              <Endpoint as ResolveEndpoint<T>>
              <aws_endpoint::EndpointShim as ResolveEndpoint<T>>
              <aws_sdk_sso::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sso::endpoint::Params>>
              <aws_sdk_sts::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sts::endpoint::Params>>
note: required by a bound in `aws_sdk_s3::config::Builder::endpoint_resolver`
   --> /Users/morganmccauley/.cargo/registry/src/github.com-1ecc6299db9ec823/aws-sdk-s3-0.23.0/src/config.rs:241:33
    |
241 |         endpoint_resolver: impl aws_smithy_http::endpoint::ResolveEndpoint<crate::endpoint::Params>
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `
aws_sdk_s3::config::Builder::endpoint_resolver`

error[E0308]: mismatched types
  --> src/types.rs:40:15
   |
9  |       let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config);
   |                         ---------------------------------------------- expected due to this value
10 |       s3_conf = s3_conf
   |  _______________^
11 | |         .endpoint_resolver(
12 | |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
13 | |         .build();
   | |________________^ expected struct `aws_sdk_s3::config::Builder`, found struct `aws_sdk_s3::Config`

error[E0308]: mismatched types
  --> src/types.rs:46:20
   |
16 |         .s3_config(s3_conf)
   |          --------- ^^^^^^^ expected struct `aws_sdk_s3::Config`, found struct `aws_sdk_s3::config::Builder`
   |          |
   |          arguments to this function are incorrect
   |
note: associated function defined here
  --> /Users/morganmccauley/Developer/Repositories/near-lake-framework-rs/src/types.rs:54:16
   |
54 |     pub(crate) s3_config: Option<aws_sdk_s3::config::Config>,
   |                ^^^^^^^^^

error[E0752]: `main` function is not allowed to be `async`
 --> src/types.rs:37:1
  |
7 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to 5 previous errors; 2 warnings emitted

Some errors have detailed explanations: E0277, E0308, E0432, E0752.
For more information about an error, try `rustc --explain E0277`.
Couldn't compile the test.
---- src/types.rs - types::LakeConfigBuilder::s3_config (line 32) stdout ----
error[E0432]: unresolved import `http`
 --> src/types.rs:34:5
  |
4 | use http::Uri;
  |     ^^^^ use of undeclared crate or module `http`

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
 --> src/types.rs:33:17
  |
3 | use aws_sdk_s3::Endpoint;
  |                 ^^^^^^^^
  |
  = note: `#[warn(deprecated)]` on by default

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
  --> src/types.rs:42:13
   |
12 |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
   |             ^^^^^^^^

error[E0277]: the trait bound `Result<Endpoint, InvalidEndpointError>: ResolveEndpoint<Params>` is not satisfied
   --> src/types.rs:42:13
    |
11  |         .endpoint_resolver(
    |          ----------------- required by a bound introduced by this call
12  |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ResolveEndpoint<Params>` is not imple
mented for `Result<Endpoint, InvalidEndpointError>`
    |
    = help: the following other types implement trait `ResolveEndpoint<Params>`:
              <&'static str as ResolveEndpoint<T>>
              <DefaultResolver as ResolveEndpoint<Params>>
              <Endpoint as ResolveEndpoint<T>>
              <aws_endpoint::EndpointShim as ResolveEndpoint<T>>
              <aws_sdk_sso::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sso::endpoint::Params>>
              <aws_sdk_sts::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sts::endpoint::Params>>
note: required by a bound in `aws_sdk_s3::config::Builder::endpoint_resolver`
   --> /Users/morganmccauley/.cargo/registry/src/github.com-1ecc6299db9ec823/aws-sdk-s3-0.23.0/src/config.rs:241:33
    |
241 |         endpoint_resolver: impl aws_smithy_http::endpoint::ResolveEndpoint<crate::endpoint::Params>
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `
aws_sdk_s3::config::Builder::endpoint_resolver`

error[E0308]: mismatched types
  --> src/types.rs:40:15
   |
9  |       let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config);
   |                         ---------------------------------------------- expected due to this value
10 |       s3_conf = s3_conf
   |  _______________^
11 | |         .endpoint_resolver(
12 | |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
13 | |         .build();
   | |________________^ expected struct `aws_sdk_s3::config::Builder`, found struct `aws_sdk_s3::Config`

error[E0308]: mismatched types
  --> src/types.rs:46:20
   |
16 |         .s3_config(s3_conf)
   |          --------- ^^^^^^^ expected struct `aws_sdk_s3::Config`, found struct `aws_sdk_s3::config::Builder`
   |          |
   |          arguments to this function are incorrect
   |
note: associated function defined here
  --> /Users/morganmccauley/Developer/Repositories/near-lake-framework-rs/src/types.rs:54:16
   |
54 |     pub(crate) s3_config: Option<aws_sdk_s3::config::Config>,
   |                ^^^^^^^^^

error[E0752]: `main` function is not allowed to be `async`
 --> src/types.rs:37:1
  |
7 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to 5 previous errors; 2 warnings emitted

Some errors have detailed explanations: E0277, E0308, E0432, E0752.
For more information about an error, try `rustc --explain E0277`.
Couldn't compile the test.
---- src/lib.rs - (line 150) stdout ----
error[E0432]: unresolved import `http`
 --> src/lib.rs:152:5
  |
4 | use http::Uri;
  |     ^^^^ use of undeclared crate or module `http`

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
 --> src/lib.rs:151:17
  |
3 | use aws_sdk_s3::Endpoint;
  |                 ^^^^^^^^
  |
  = note: `#[warn(deprecated)]` on by default

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
  --> src/lib.rs:160:13
   |
12 |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
   |             ^^^^^^^^

error[E0277]: the trait bound `Result<Endpoint, InvalidEndpointError>: ResolveEndpoint<Params>` is not satisfied
   --> src/lib.rs:160:13
    |
11  |     .endpoint_resolver(
    |      ----------------- required by a bound introduced by this call
12  |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ResolveEndpoint<Params>` is not imple
mented for `Result<Endpoint, InvalidEndpointError>`
    |
    = help: the following other types implement trait `ResolveEndpoint<Params>`:
              <&'static str as ResolveEndpoint<T>>
              <DefaultResolver as ResolveEndpoint<Params>>
              <Endpoint as ResolveEndpoint<T>>
              <aws_endpoint::EndpointShim as ResolveEndpoint<T>>
              <aws_sdk_sso::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sso::endpoint::Params>>
              <aws_sdk_sts::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sts::endpoint::Params>>
note: required by a bound in `aws_sdk_s3::config::Builder::endpoint_resolver`
   --> /Users/morganmccauley/.cargo/registry/src/github.com-1ecc6299db9ec823/aws-sdk-s3-0.23.0/src/config.rs:241:33
    |
241 |         endpoint_resolver: impl aws_smithy_http::endpoint::ResolveEndpoint<crate::endpoint::Params>
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `
aws_sdk_s3::config::Builder::endpoint_resolver`

error[E0308]: mismatched types
  --> src/lib.rs:158:11
   |
9  |   let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config);
   |                     ---------------------------------------------- expected due to this value
10 |   s3_conf = s3_conf
   |  ___________^
11 | |     .endpoint_resolver(
12 | |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
13 | |     .build();
   | |____________^ expected struct `aws_sdk_s3::config::Builder`, found struct `aws_sdk_s3::Config`

error[E0308]: mismatched types
  --> src/lib.rs:164:16
   |
16 |     .s3_config(s3_conf)
   |      --------- ^^^^^^^ expected struct `aws_sdk_s3::Config`, found struct `aws_sdk_s3::config::Builder`
   |      |
   |      arguments to this function are incorrect
   |
note: associated function defined here
  --> /Users/morganmccauley/Developer/Repositories/near-lake-framework-rs/src/types.rs:54:16
   |
54 |     pub(crate) s3_config: Option<aws_sdk_s3::config::Config>,
   |                ^^^^^^^^^

error[E0752]: `main` function is not allowed to be `async`
 --> src/lib.rs:155:1
  |
7 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to 5 previous errors; 2 warnings emitted

Some errors have detailed explanations: E0277, E0308, E0432, E0752.
For more information about an error, try `rustc --explain E0277`.
Couldn't compile the test.
---- src/types.rs - types::LakeConfigBuilder::s3_config (line 32) stdout ----
error[E0432]: unresolved import `http`
 --> src/types.rs:34:5
  |
4 | use http::Uri;
  |     ^^^^ use of undeclared crate or module `http`

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
 --> src/types.rs:33:17
  |
3 | use aws_sdk_s3::Endpoint;
  |                 ^^^^^^^^
  |
  = note: `#[warn(deprecated)]` on by default

warning: use of deprecated struct `aws_sdk_s3::Endpoint`: Use `.endpoint_url(...)` directly instead
  --> src/types.rs:42:13
   |
12 |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
   |             ^^^^^^^^

error[E0277]: the trait bound `Result<Endpoint, InvalidEndpointError>: ResolveEndpoint<Params>` is not satisfied
   --> src/types.rs:42:13
    |
11  |         .endpoint_resolver(
    |          ----------------- required by a bound introduced by this call
12  |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ResolveEndpoint<Params>` is not imple
mented for `Result<Endpoint, InvalidEndpointError>`
    |
    = help: the following other types implement trait `ResolveEndpoint<Params>`:
              <&'static str as ResolveEndpoint<T>>
              <DefaultResolver as ResolveEndpoint<Params>>
              <Endpoint as ResolveEndpoint<T>>
              <aws_endpoint::EndpointShim as ResolveEndpoint<T>>
              <aws_sdk_sso::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sso::endpoint::Params>>
              <aws_sdk_sts::endpoint::DefaultResolver as ResolveEndpoint<aws_sdk_sts::endpoint::Params>>
note: required by a bound in `aws_sdk_s3::config::Builder::endpoint_resolver`
   --> /Users/morganmccauley/.cargo/registry/src/github.com-1ecc6299db9ec823/aws-sdk-s3-0.23.0/src/config.rs:241:33
    |
241 |         endpoint_resolver: impl aws_smithy_http::endpoint::ResolveEndpoint<crate::endpoint::Params>
    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in 
aws_sdk_s3::config::Builder::endpoint_resolver`

error[E0308]: mismatched types
  --> src/types.rs:40:15
   |
9  |       let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config);
   |                         ---------------------------------------------- expected due to this value
10 |       s3_conf = s3_conf
   |  _______________^
11 | |         .endpoint_resolver(
12 | |             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
13 | |         .build();
   | |________________^ expected struct `aws_sdk_s3::config::Builder`, found struct `aws_sdk_s3::Config`

error[E0308]: mismatched types
  --> src/types.rs:46:20
   |
16 |         .s3_config(s3_conf)
   |          --------- ^^^^^^^ expected struct `aws_sdk_s3::Config`, found struct `aws_sdk_s3::config::Builder`
   |          |
   |          arguments to this function are incorrect
   |
note: associated function defined here
  --> /Users/morganmccauley/Developer/Repositories/near-lake-framework-rs/src/types.rs:54:16
   |
54 |     pub(crate) s3_config: Option<aws_sdk_s3::config::Config>,
   |                ^^^^^^^^^

error[E0752]: `main` function is not allowed to be `async`
 --> src/types.rs:37:1
  |
7 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to 5 previous errors; 2 warnings emitted

Some errors have detailed explanations: E0277, E0308, E0432, E0752.
For more information about an error, try `rustc --explain E0277`.
Couldn't compile the test.
---- src/types.rs - types::LakeConfigBuilder::testnet (line 79) stdout ----
error[E0752]: `main` function is not allowed to be `async`
 --> src/types.rs:82:1
  |
5 | async fn main() {
  | ^^^^^^^^^^^^^^^ `main` function is not allowed to be `async`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0752`.
Couldn't compile the test.

failures:
    src/lib.rs - (line 103)
    src/lib.rs - (line 115)
    src/lib.rs - (line 150)
    src/lib.rs - (line 7)
    src/lib.rs - (line 73)
    src/lib.rs - streamer (line 241)
    src/types.rs - types::LakeConfig (line 7)
    src/types.rs - types::LakeConfig::s3_config (line 32)
    src/types.rs - types::LakeConfigBuilder::betanet (line 97)
    src/types.rs - types::LakeConfigBuilder::mainnet (line 61)
    src/types.rs - types::LakeConfigBuilder::s3_config (line 32)
    src/types.rs - types::LakeConfigBuilder::s3_config (line 32)
    src/types.rs - types::LakeConfigBuilder::testnet (line 79)

test result: FAILED. 0 passed; 13 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.81s

error: doctest failed, to rerun pass `--doc`

```
</details>

--- a/lake-framework/README.md
+++ b/lake-framework/README.md
@@ -6,49 +6,26 @@ the NEAR Protocol data.
 
 ## Example
 
-```ignore
-use futures::StreamExt;
-use near_lake_framework::LakeConfigBuilder;
-
-#[tokio::main]
-async fn main() -> Result<(), tokio::io::Error> {
-    // create a NEAR Lake Framework config
-    let config = LakeConfigBuilder::default()
+```no_run
+fn main() -> anyhow::Result<()> {
+    near_lake_framework::LakeBuilder::default()
         .testnet()
-        .start_block_height(82422587)
-        .build()
-        .expect("Failed to build LakeConfig");
-
-    // instantiate the NEAR Lake Framework Stream
-    let (sender, stream) = near_lake_framework::streamer(config);
-
-    // read the stream events and pass them to a handler function with
-    // concurrency 1
-    let mut handlers = tokio_stream::wrappers::ReceiverStream::new(stream)
-        .map(|streamer_message| handle_streamer_message(streamer_message))
-        .buffer_unordered(1usize);
-
-    while let Some(_handle_message) = handlers.next().await {}
-    drop(handlers); // close the channel so the sender will stop
-
-    // propagate errors from the sender
-    match sender.await {
-        Ok(Ok(())) => Ok(()),
-        Ok(Err(e)) => Err(e),
-        Err(e) => Err(anyhow::Error::from(e)), // JoinError
-    }
+        .start_block_height(112205773)
+        .build()?
+        .run(handle_block)
 }
 
-// The handler function to take the entire `StreamerMessage`
-// and print the block height and number of shards
-async fn handle_streamer_message(
-    streamer_message: near_lake_framework::near_indexer_primitives::StreamerMessage,
-) {
+// The handler function to take the `Block`
+// and print the block height
+async fn handle_block(
+    block: near_lake_primitives::block::Block,
+    _context: near_lake_framework::LakeContext,
+) -> anyhow::Result<()> {
     eprintln!(
-        "{} / shards {}",
-        streamer_message.block.header.height,
-        streamer_message.shards.len()
+        "Block #{}",
+        block.block_height(),
     );
+#    Ok(())
 }
 ```
 
@@ -73,10 +50,10 @@ In order to be able to get objects from the AWS S3 bucket you need to provide th
 
 #### Passing credentials to the config builder
 
-```rust
+```
 use near_lake_framework::LakeBuilder;
 
-# async fn main() {
+# fn main() {
 let credentials = aws_credential_types::Credentials::new(
     "AKIAIOSFODNN7EXAMPLE",
     "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
@@ -88,9 +65,10 @@ let s3_config = aws_sdk_s3::Config::builder()
     .credentials_provider(credentials)
     .build();
 
-let config = LakeBuilder::default()
+let lake = LakeBuilder::default()
     .s3_config(s3_config)
     .s3_bucket_name("near-lake-data-custom")
+    .s3_region_name("eu-central-1")
     .start_block_height(1)
     .build()
     .expect("Failed to build LakeConfig");
@@ -125,7 +103,7 @@ $ AWS_DEFAULT_REGION=eu-central-1
 
 Add the following dependencies to your `Cargo.toml`
 
-```toml
+```text
 ...
 [dependencies]
 futures = "0.3.5"
@@ -134,7 +112,7 @@ tokio = { version = "1.1", features = ["sync", "time", "macros", "rt-multi-threa
 tokio-stream = { version = "0.1" }
 
 # NEAR Lake Framework
-near-lake-framework = "0.6.1"
+near-lake-framework = "0.8.0"
 ```
 
 ### Custom S3 storage
@@ -148,24 +126,22 @@ In case you want to run your own [near-lake](https://github.com/near/near-lake) 
 $ mkdir -p /data/near-lake-custom && minio server /data
 ```
 
- - pass custom `aws_sdk_s3::config::Config` to the [LakeConfigBuilder]
+ - pass custom `aws_sdk_s3::config::Config` to the [LakeBuilder]
 
-```rust
-use aws_sdk_s3::Endpoint;
-use http::Uri;
+```
 use near_lake_framework::LakeBuilder;
 
-# fn main() {
+# #[tokio::main]
+# async fn main() {
 let aws_config = aws_config::from_env().load().await;
-let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config);
-s3_conf = s3_conf
-    .endpoint_resolver(
-        Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
+let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config)
+    .endpoint_url("http://0.0.0.0:9000")
     .build();
 
-let config = LakeBuilder::default()
+let lake = LakeBuilder::default()
     .s3_config(s3_conf)
     .s3_bucket_name("near-lake-data-custom")
+    .s3_region_name("eu-central-1")
     .start_block_height(1)
     .build()
     .expect("Failed to build LakeConfig");

--- a/lake-framework/README.md
+++ b/lake-framework/README.md
@@ -50,7 +50,7 @@ In order to be able to get objects from the AWS S3 bucket you need to provide th
 
 #### Passing credentials to the config builder
 
-```
+```rust
 use near_lake_framework::LakeBuilder;
 
 # fn main() {
@@ -103,7 +103,7 @@ $ AWS_DEFAULT_REGION=eu-central-1
 
 Add the following dependencies to your `Cargo.toml`
 
-```text
+```toml
 ...
 [dependencies]
 futures = "0.3.5"
@@ -122,7 +122,7 @@ In case you want to run your own [near-lake](https://github.com/near/near-lake) 
 
  - run minio
 
-```text
+```bash
 $ mkdir -p /data/near-lake-custom && minio server /data
 ```
 

--- a/lake-framework/src/lib.rs
+++ b/lake-framework/src/lib.rs
@@ -16,24 +16,17 @@ pub(crate) mod types;
 pub(crate) const LAKE_FRAMEWORK: &str = "near_lake_framework";
 
 /// Creates `mpsc::channel` and returns the `receiver` to read the stream of `StreamerMessage`
-/// ```
-/// use near_lake_framework::LakeConfigBuilder;
-/// use tokio::sync::mpsc;
-///
-/// # async fn main() {
-///    let config = LakeConfigBuilder::default()
+///```no_run
+///# fn main() -> anyhow::Result<()> {
+///    near_lake_framework::LakeBuilder::default()
 ///        .testnet()
-///        .start_block_height(82422587)
-///        .build()
-///        .expect("Failed to build LakeConfig");
+///        .start_block_height(112205773)
+///        .build()?
+///        .run(handle_block)
+///# }
 ///
-///     let (_, stream) = near_lake_framework::streamer(config);
-///
-///     while let Some(streamer_message) = stream.recv().await {
-///         eprintln!("{:#?}", streamer_message);
-///     }
-/// # }
-/// ```
+/// # async fn handle_block(_block: near_lake_primitives::block::Block, _context: near_lake_framework::LakeContext) -> anyhow::Result<()> { Ok(()) }
+///```
 impl types::Lake {
     pub fn run<Fut>(
         self,

--- a/lake-framework/src/types.rs
+++ b/lake-framework/src/types.rs
@@ -2,17 +2,17 @@
 pub type BlockHeight = u64;
 
 /// Configuration struct for NEAR Lake Framework
-/// NB! Consider using [`LakeConfigBuilder`]
-/// Building the `LakeConfig` example:
+/// NB! Consider using [`LakeBuilder`]
+/// Building the `Lake` example:
 /// ```
-/// use near_lake_framework::LakeConfigBuilder;
+/// use near_lake_framework::LakeBuilder;
 ///
-/// # async fn main() {
-///    let config = LakeConfigBuilder::default()
+/// # fn main() {
+///    let lake = LakeBuilder::default()
 ///        .testnet()
 ///        .start_block_height(82422587)
 ///        .build()
-///        .expect("Failed to build LakeConfig");
+///        .expect("Failed to build Lake");
 /// # }
 /// ```
 #[derive(Default, Builder, Debug)]
@@ -30,24 +30,22 @@ pub struct Lake {
     /// ## Use-case: custom endpoint
     /// You might want to stream data from the custom S3-compatible source () . In order to do that you'd need to pass `aws_sdk_s3::config::Config` configured
     /// ```
-    /// use aws_sdk_s3::Endpoint;
-    /// use http::Uri;
-    /// use near_lake_framework::LakeConfigBuilder;
+    /// use near_lake_framework::LakeBuilder;
     ///
+    /// # #[tokio::main]
     /// # async fn main() {
     ///     let aws_config = aws_config::from_env().load().await;
-    ///     let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config);
-    ///     s3_conf = s3_conf
-    ///         .endpoint_resolver(
-    ///             Endpoint::immutable("http://0.0.0.0:9000".parse::<Uri>().unwrap()))
+    ///     let mut s3_conf = aws_sdk_s3::config::Builder::from(&aws_config)
+    ///         .endpoint_url("http://0.0.0.0:9000")
     ///         .build();
     ///
-    ///     let config = LakeConfigBuilder::default()
+    ///     let lake = LakeBuilder::default()
     ///         .s3_config(s3_conf)
     ///         .s3_bucket_name("near-lake-data-custom")
+    ///         .s3_region_name("eu-central-1")
     ///         .start_block_height(1)
     ///         .build()
-    ///         .expect("Failed to build LakeConfig");
+    ///         .expect("Failed to build Lake");
     /// # }
     /// ```
     #[builder(setter(strip_option), default)]
@@ -57,16 +55,16 @@ pub struct Lake {
 }
 
 impl LakeBuilder {
-    /// Shortcut to set up [LakeConfigBuilder::s3_bucket_name] for mainnet
+    /// Shortcut to set up [LakeBuilder::s3_bucket_name] for mainnet
     /// ```
-    /// use near_lake_framework::LakeConfigBuilder;
+    /// use near_lake_framework::LakeBuilder;
     ///
-    /// # async fn main() {
-    ///    let config = LakeConfigBuilder::default()
+    /// # fn main() {
+    ///    let lake = LakeBuilder::default()
     ///        .mainnet()
     ///        .start_block_height(65231161)
     ///        .build()
-    ///        .expect("Failed to build LakeConfig");
+    ///        .expect("Failed to build Lake");
     /// # }
     /// ```
     pub fn mainnet(mut self) -> Self {
@@ -75,16 +73,16 @@ impl LakeBuilder {
         self
     }
 
-    /// Shortcut to set up [LakeConfigBuilder::s3_bucket_name] for testnet
+    /// Shortcut to set up [LakeBuilder::s3_bucket_name] for testnet
     /// ```
-    /// use near_lake_framework::LakeConfigBuilder;
+    /// use near_lake_framework::LakeBuilder;
     ///
-    /// # async fn main() {
-    ///    let config = LakeConfigBuilder::default()
+    /// # fn main() {
+    ///    let lake = LakeBuilder::default()
     ///        .testnet()
     ///        .start_block_height(82422587)
     ///        .build()
-    ///        .expect("Failed to build LakeConfig");
+    ///        .expect("Failed to build Lake");
     /// # }
     /// ```
     pub fn testnet(mut self) -> Self {
@@ -93,16 +91,16 @@ impl LakeBuilder {
         self
     }
 
-    /// Shortcut to set up [LakeConfigBuilder::s3_bucket_name] for betanet
+    /// Shortcut to set up [LakeBuilder::s3_bucket_name] for betanet
     /// ```
-    /// use near_lake_framework::LakeConfigBuilder;
+    /// use near_lake_framework::LakeBuilder;
     ///
-    /// # async fn main() {
-    ///    let config = LakeConfigBuilder::default()
+    /// # fn main() {
+    ///    let lake = LakeBuilder::default()
     ///        .betanet()
     ///        .start_block_height(82422587)
     ///        .build()
-    ///        .expect("Failed to build LakeConfig");
+    ///        .expect("Failed to build Lake");
     /// # }
     /// ```
     pub fn betanet(mut self) -> Self {

